server/src/test/java/com/adobe/testing/s3mock/its/ListObjectIT.java

```diff
@@ -0,0 +1,194 @@
+
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+
+import com.adobe.testing.s3mock.S3MockApplication;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.logging.log4j.util.Strings;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(Parameterized.class)
+public class ListObjectIT {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ListObjectIT.class);
+
+  private static final String BUCKET_NAME = "list-objects-test";
+
+  private static final String[] ALL_OBJECTS = 
+      arr("a", "b", "b/1", "b/1/1", "b/1/2", "b/2", "c/1", "c/1/1");
+
+  private static final boolean RUN_AGAINST_AWS = false;
+  
+  private static <T> T[] arr(@SuppressWarnings("unchecked") final T... a) {
+    return a;
+  }
+
+  /**
+   * Parameter fatcory.
+   * 
+   * @return
+   */
+  @Parameters(name = "{index}: prefix={0}, delimiter={1}")
+  public static Iterable<Object[]> data() {
+    return Arrays.<Object[]>asList(//
+        arr(null, null, ALL_OBJECTS, arr()), //
+        arr("", null, ALL_OBJECTS, arr()), //
+        arr(null, "", ALL_OBJECTS, arr()), //
+        arr("/", null, arr(), arr()), //
+        arr("b", null, arr("b", "b/1", "b/1/1", "b/1/2", "b/2"), arr()), //
+        arr("b/", null, arr("b/1", "b/1/1", "b/1/2", "b/2"), arr()), //
+        arr("b", "/", arr("b"), arr("b/")), //
+        arr("b/", "/", arr("b/1", "b/2"), arr("b/1/")), //
+        arr("b/1", "/", arr("b/1"), arr("b/1/")), //
+        arr("b/1/", "/", arr("b/1/1", "b/1/2"), arr()), //
+        arr("c", "/", arr(), arr("c/")), //
+        arr("c/", "/", arr("c/1"), arr("c/1/")) //
+    );
+  }
+
+  private static final List<String> param = new ArrayList<>();
+
+  @Parameter(0)
+  public String prefix;
+
+  @Parameter(1)
+  public String delimiter;
+
+  @Parameter(2)
+  public Object[] expectedKeys;
+
+  @Parameter(3)
+  public Object[] expectedPrefixes;
+
+
+  private static S3MockApplication s3mock;

```

Comment:
The S3Mock is already expected to run at this point (as Docker container in the build or started manually in the IDE with default settings). I'd prefer we keep it with the same expectations here as well.

Reply:
In fact, I was wondering about why you would opt for launching a separate container as opposed to just doing what this library is well suited to do: run it in embedded mode. Not having to deal with containers or manually started processes makes things so much easier.

Reply:
The ITs in this Maven module are supposed to verify the stand-alone app, and that's done through the Docker image here.

Testing the embedded mode is primarily done through the various `testsupport` modules.

I agree with the manual starting of the other process being a bit inconvenient... it's a trade-off, but in the end I find it a bit more important to have consistency in the ITs of a single module.